src/index.js

```diff
@@ -1,10 +1,167 @@
 import { months } from './data.js';
 
 (async function () {
-  const allCityDetails = await fetchCityDetails()
+  let allCityDetails = await fetchCityDetails()
+  const icons = document.querySelectorAll('.mid-ic')
+  const leftArrow = document.getElementById('left-scroll-arrow')
+  const rightArrow = document.getElementById('right-scroll-arrow')
+  let middleSegmentCards = allCityDetails
+  let selectedIcon = ' '
+  let cardCount = document.getElementById('select-icon').value
+  allCityDetails = sortCityOptions(allCityDetails)
   loadSelectOptions(allCityDetails)
   changeHeaderValues()
   document.querySelector('.option-select.city').addEventListener('change', changeHeaderValues)
+  changeMiddleSegment(allCityDetails)
+  getScrollWidth()
+  icons.forEach((ic) => ic.addEventListener('click', selectIcon))
+  leftArrow.addEventListener('click', () => { scrollItems('left') })
+  rightArrow.addEventListener('click', () => { scrollItems('right') })
+  document.getElementById('select-icon').addEventListener('change', getCardCount)
+
+  /**
+   *  This function will get the scrollwidth and clientwidth of the itemscontainer to display the arrow
+   */
+  function getScrollWidth () {
+    const middleContainerWidth = document.getElementsByClassName('items-container')[0].clientWidth
+    const scrollContainerWidth = document.getElementsByClassName('items-container')[0].scrollWidth
+    if (middleContainerWidth >= scrollContainerWidth) {
+      document.getElementsByClassName('left-arr')[0].style.display = 'none'
+      document.getElementsByClassName('right-arr')[0].style.display = 'none'
+      document.getElementsByClassName('items-container')[0].style.justifyContent = 'space-around'
+    } else {
+      document.getElementsByClassName('left-arr')[0].style.display = 'block'
+      document.getElementsByClassName('right-arr')[0].style.display = 'block'
+    }
+  }
+
+  /**
+   * This function will get the value from spinner  and update the card numbers
+   */
+  function getCardCount () {
+    cardCount = document.getElementById('select-icon').value
+    changeMiddleSegment(middleSegmentCards)
+    getScrollWidth()
```

Comment:
the function names like get... will be used whenever it is returning something.
Here as it's not returning some data and directly updating the arrows display on scroll width we can name it something like `updateMiddleCitiesScrollButtonVisibility`. It is fine if the function name is a little large until it conveys its purpose.